src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -266,19 +265,28 @@ private static function normalizeBody($body)
     private static function normalizePeerFingerprint($fingerprint): array
     {
         if (\is_string($fingerprint)) {
-            switch (\strlen($fingerprint = str_replace(':', '', $fingerprint))) {
-                case 32: $fingerprint = ['md5' => $fingerprint]; break;
-                case 40: $fingerprint = ['sha1' => $fingerprint]; break;
-                case 44: $fingerprint = ['pin-sha256' => [$fingerprint]]; break;
-                case 64: $fingerprint = ['sha256' => $fingerprint]; break;
-                default: throw new InvalidArgumentException(sprintf('Cannot auto-detect fingerprint algorithm for "%s".', $fingerprint));
+            $algos_strlen_mapping = [
+                '32' => 'md5',
+                '40' => 'sha1',
+                '44' => 'pin-sha256',
+                '64' => 'sha256',
+            ];
+
+            $length = \strlen($fingerprint = str_replace(':', '', $fingerprint));
+
+            if (!isset($algos_strlen_mapping[$length])) {
+                throw new InvalidArgumentException(sprintf('Cannot auto-detect fingerprint algorithm for "%s".', $fingerprint));
             }
+
+            $fingerprint = [
+                $algos_strlen_mapping[$length] => $fingerprint,
```

Comment:
When using `pin-sha256` the value should be an array.

Reply:
Fixed.